collector/textfile.go

```diff
@@ -52,31 +65,165 @@ func NewTextFileCollector() (Collector, error) {
 	c := &textFileCollector{
 		path: *textFileDirectory,
 	}
+	return c, nil
+}
 
-	if c.path == "" {
-		// This collector is enabled by default, so do not fail if
-		// the flag is not passed.
-		log.Infof("No directory specified, see --collector.textfile.directory")
-	} else {
-		textFileAddOnce.Do(func() {
-			prometheus.DefaultGatherer = prometheus.Gatherers{
-				prometheus.DefaultGatherer,
-				prometheus.GathererFunc(func() ([]*dto.MetricFamily, error) { return c.parseTextFiles(), nil }),
+func convertMetricFamilies(metricFamilies []*dto.MetricFamily, ch chan<- prometheus.Metric) {
+	var valType prometheus.ValueType
+	var val float64
+
+	for _, mf := range metricFamilies {
+		labelNames := map[string]struct{}{}
+		for _, metric := range mf.Metric {
+			labelPairs := metric.GetLabel()
```

Comment:
Some naming suggestions to be consistent with our usual nomenclature: a label is the whole key/value pair. A label has a label name and a label value. I would probably just call `labelPairs` -> `labels` instead.

Reply:
(and yeah, there'll be some instances where we call labels label pairs, but just label is more common)